app/server/lib/DocApi.ts

```diff
@@ -1536,6 +1534,22 @@ export class DocWorkerApi {
       return res.status(200).json(docId);
     }));
 
+    this._app.post('/api/docs/:docId/copy', canView, expressWrap(async (req, res) => {
+      const userId = getUserId(req);
+
+      const parameters: {[key: string]: any} = req.body;
+
+      const docId = await this._copyDocToWorkspace(req, {
+        userId,
+        sourceDocumentId: stringParam(req.params.docId, 'docId'),
+        workspaceId: integerParam(parameters.workspaceId, 'workspaceId'),
+        documentName: stringParam(parameters.documentName, 'documentName'),
+        asTemplate: optBooleanParam(parameters.asTemplate, 'asTemplate'),
+      });
```

Comment:
If this throws (say, out of disk space), how will the error be reported to the user?

Reply:
Probably the raw error if it's a disk space error?. Looks like the underlying _copyToWorkspace function handles some errors, but not all, so whatever error is thrown probably propagates to the topmost handler.

That said, I think fixing this is probably a larger question, because:
1. This is what the /docs endpoint already does.
2. This is a common theme across our endpoints, so it's probably a wider discussion we should have.

Reply:
Right, my question was if we should specifically handle the error here or not. Sounds like that should be done elsewhere anyway. That works for me.